core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -219,6 +221,10 @@ final class ShuffleBlockFetcherIterator(
         logWarning("Failed to cleanup shuffle fetch temp file " + file.path())
       }
     }
+    // Null out the referent in cleanup listener to make sure we don't keep a reference
+    // to this ShuffleBlockFetcherIterator, after we're done reading from it, to let it be
```

Comment:
can you point out where `ShuffleBlockFetcherIterator` occupies a large amount of memory?

Reply:
@cloud-fan it's here 

> Otherwise we can metadata on block locations

Reply:
Do you mean the `fetchRequests`? Can we call `fetchRequests.clear` to release the memory?

Reply:
@cloud-fan No, it's the `blocksByAddress` in the constructor of `ShuffleBlockFetcherIterator`. Let me make it more clear in the PR description and this comment.

Reply:
Shall we simply mark `blocksByAddress` as `var` and null it out in `cleanup`? Then it becomes a simple 2-line change.